l2geth/eth/api_backend.go

```diff
@@ -203,10 +203,12 @@ func (b *EthAPIBackend) BlockByNumberOrHash(ctx context.Context, blockNrOrHash r
 }
 
 func (b *EthAPIBackend) StateAndHeaderByNumber(ctx context.Context, number rpc.BlockNumber) (*state.StateDB, *types.Header, error) {
-	// Pending state is only known by the miner
-	if number == rpc.PendingBlockNumber {
-		block, state := b.eth.miner.Pending()
-		return state, block.Header(), nil
+	if !vm.UsingOVM {
+		// Pending state is only known by the miner
```

Comment:
This probably should be hidden still

https://github.com/ethereum-optimism/optimism/blob/8082d16c306aab1e53a019ecbce7038d2f1415ca/l2geth/rpc/types.go#L65-L68